### PR TITLE
#158295488 Setup graphql and apollo client

### PR DESCRIPTION
**Pull request by malmike**
#### What Does This PR Do?
- Setup graphql and apollo client

#### Description Of Task To Be Completed
- Add ApolloProvider to wrap around the React app and place the client on the context
- Install the other dependencies necessary for graphql and apollo to run

#### Any Background Context You Want To Provide?
- Moving from using rest to graphql to handle api calls

#### How can this be manually tested?
- Clone the project and run the application based on the specifications in the readme

#### What are the relevant github issues?
[#158295488](https://www.pivotaltracker.com/story/show/158295488)

### CODE REVIEW

**Comment by nwochaadim** [comment]
Please create a WIP label and attach a frontend label to this PR

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "author": "",
   "license": "MIT",
   "dependencies": {
+    "apollo-client-preset": "^1.0.8",
```
Comment (andela-sjames) [inline package.json]: This should be locked(version should be pinned)

File: client/store/configureStore.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
 import thunk from 'redux-thunk';
 import rootReducer from '../reducers/index';
 import { persistStore, persistCombineReducers } from 'redux-persist';
 import storage from 'redux-persist/es/storage';
 import { saveTokenMiddleware } from '../middleware/auth';
+import { ApolloClient } from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+
+const httpLink = new HttpLink({uri: 'http://localhost:9000'})
```
Comment (andela-sjames) [inline client/store/configureStore.js]: URI should not be hardcoded here

File: client/client.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { HttpLink } from 'apollo-link-http';
+import { ApolloLink } from 'apollo-link';
+
+const cache = new InMemoryCache();
+const httpLink = new HttpLink({
+    uri: process.env.API_URI
+});
+
+const middlewareAuthLink = new ApolloLink((operation, forward) => {
+    const token = "token"
```
Comment (nwochaadim) [inline client/client.js]: "token" should not be hardcoded but retrieved via an env variable. Can you remind me why we need this `middlewareAuthLink`?

File: client/client.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    const authorizationHeader = token ? token : null
```
[the remark this answers is not in the document]
Comment (malmike) [reply client/client.js]: So middlewareAuth is used to allow header information be added to the httpLink uri dynamically. As for the reason token was hard coded, I was not sure whether the token would be picked from the redux store or the browser storage options

File: client/package.json [hunk 5]
```diff
@@ -34,12 +35,15 @@
     "extract-text-webpack-plugin": "3.0.0",
     "file-loader": "0.11.2",
     "html-webpack-plugin": "^3.2.0",
+    "graphql": "^0.13.2",
+    "graphql-tag": "2.9.2",
```
Comment (nwochaadim) [inline client/package.json]: Apollo should provide the `gql`. so we don't need this

File: client/package.json [hunk 6]
```diff
@@ -19,6 +19,7 @@
   "author": "",
   "license": "MIT",
   "dependencies": {
+    "apollo-client-preset": "1.0.8",
```
Comment (nwochaadim) [inline client/package.json]: this package has been deprecated. we should not use this. Rather use `apollo-boost` so we don't have issues in the future

File: .env.sample [hunk 7]
```diff
@@ -18,3 +18,5 @@ DJANGO_SETTINGS_MODULE=
 NODE_ENV=
 DEBUG=
 ENVIRONMENT=
+REACT_APP_ACCESS_TOKEN="551b10b0c4164b58f566889e0a6b1880aa4680ae"
+REACT_APP_API_URI="https://localhost:8000/graphql"
```
Comment (nwochaadim) [inline .env.sample]: Add a trailing line here. You can actually configure your editor to automatically add this for you

File: client/client.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+const cache = new InMemoryCache();
+const httpLink = new HttpLink({
+    uri: process.env.API_URI
+});
+
+const Client = new ApolloClient({
+    link: httpLink,
+    cache: cache
+})
```
Comment (nwochaadim) [inline client/client.js]: you missed a semicolon here

File: client/webpack.config.dev.js [hunk 9]
```diff
@@ -11,57 +11,81 @@ module.exports = {
...
     loaders: [
-      { test: /\.(js|jsx)$/, exclude: /node_modules/, loader: 'babel-loader', query: { presets: ['es2015', 'react'] } },
-      { test: /(\.css|scss)$/, loaders: ['style-loader', 'css-loader', 'sass-loader'] },
-      { test: /\.(jpeg|png|gif|svg|jpg)$/, loader: 'url-loader?limit=25000' },
-      { test: /\.eot(\?v=\d+\.\d+\.\d+)?$/, loader: 'file-loader' },
-      { test: /\.(woff|woff2)$/, loader: 'url?prefix=font/&limit=5000' },
-      { test: /\.ttf(\?v=\d+\.\d+\.\d+)?$/, loader: 'url?limit=10000&mimetype=application/octet-stream' },
-      { test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: 'url?limit=10000&mimetype=image/svg+xml' }
-    ],
+      {
```
Comment (nwochaadim) [inline client/webpack.config.dev.js]: this has nothing to do with the changes or was there a reason you had to edit this? if not, you can revert this aspect back

File: .env.sample [hunk 10]
```diff
@@ -1,20 +1,21 @@
...
 NODE_ENV=
 DEBUG=
+CLIENT_ID=
+CLIENT_SECRET=
+SLACK_BOT_TOKEN=
+SECRET_KEY=
 ENVIRONMENT=
+SLACK_TOKEN_URL=
+G_SUITE_DOMAIN=andela.com
+REACT_APP_API_URI="https://localhost:8000/graphql"
```
Comment (nwochaadim) [inline .env.sample]: i dont think `https` will work. just leave it as `http`

**Pull request merged**